### PR TITLE
(BKR-865) Modularize vmpooler hypervisor

### PR DESCRIPTION
**Pull request by rishijavia**
Separate vmpooler hypervisor to an external module: [beaker-vmpooler](https://github.com/rishijavia/beaker-vmpooler).

- [x] Review and publish beaker [beaker-vmpooler](https://github.com/rishijavia/beaker-vmpooler)
- [x] CI should pass on this PR once the module is published
- [ ] Review and merge this PR

[ci skip]

### CODE REVIEW

File: beaker.gemspec [hunk 1]
```diff
@@ -56,6 +56,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency 'aws-sdk-v1', '~> 1.57'
   s.add_runtime_dependency 'docker-api'
   s.add_runtime_dependency 'fog', '~> 1.38'
+  s.add_runtime_dependency 'beaker-vmpooler'
```
Comment (tvpartytonight) [inline beaker.gemspec]: This should be versioned, probably like ` '~> 0.1'`

File: lib/beaker/hypervisor.rb [hunk 2]
```diff
@@ -150,6 +148,6 @@ def generate_host_name
   end
 end
 
-[ 'vsphere_helper', 'vagrant', 'vagrant_custom', 'vagrant_virtualbox', 'vagrant_parallels', 'vagrant_libvirt', 'vagrant_fusion', 'vagrant_workstation', 'fusion', 'aws_sdk', 'vsphere', 'vmpooler', 'vcloud', 'docker', 'google_compute', 'openstack', 'noop' ].each do |lib|
+[ 'vsphere_helper', 'vagrant', 'vagrant_custom', 'vagrant_virtualbox', 'vagrant_parallels', 'vagrant_libvirt', 'vagrant_fusion', 'vagrant_workstation', 'fusion', 'aws_sdk', 'vsphere', 'vcloud', 'docker', 'google_compute', 'openstack', 'noop' ].each do |lib|
```
Comment (tvpartytonight) [inline lib/beaker/hypervisor.rb]: Per conversation, this should be expanded to a list instead of one single line.

**Comment by rishijavia** [comment]
[ci skip]

**Comment by kevpl** [comment]
@rishijavia you're going to have to cleanup your git history for this PR. What's up with the revert commit?

**Comment by kevpl** [comment]
@rishijavia looks like a transient. The BFA caught it, so we're good to re-kick, so I'll do that now.

@puppetlabs-jenkins retest this please

**Comment by rishijavia** [comment]
@kevpl similar error, was not able to checkout PR ref in one of the [CI pipelines](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-rspec-integration/2570/agent=ubuntu-12.04-amd64___ubuntu-1204-x86_64,master=ubuntu-12.04-amd64___ubuntu-1204-x86_64/console). It was able to do the same in [other pipelines](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_unit-rb_rspec/4570/ruby=2.2.5/consoleFull), so i take its another transient failure. Kicking CI again.

@puppetlabs-jenkins retest this please

**Pull request merged**